schema.go

```diff
@@ -28,6 +28,7 @@ type ISchema interface {
 
 type Schema[T any] struct {
 	validators []Validator[T]
+	isOptional *bool
```

Comment:
```suggestion
	isOptional bool
```

Is there a specific need why this needs to be a pointer, if we specify it was `bool`, so it's not nullable, the default value should be `false`.

Reply:
makes things wayyy easier <3